core/primitives/src/receipt.rs

```diff
@@ -98,3 +101,125 @@ impl fmt::Debug for DataReceipt {
             .finish()
     }
 }
+
+#[derive(BorshSerialize, BorshDeserialize, Hash, PartialEq, Eq, Clone)]
+pub struct ReceivedData {
+    pub data: Option<Vec<u8>>,
+}
+
+impl fmt::Debug for ReceivedData {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        f.debug_struct("ReceivedData")
+            .field("data", &format_args!("{}", logging::pretty_result(&self.data)))
+            .finish()
+    }
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug)]
+pub struct ReceiptView {
```

Comment:
i was trying to keep Views in one file. We can rename rpc.rs to views.rs or something

Reply:
renamed and moved